### PR TITLE
Add table aws_cloudtrail_event_data_store closes #1423

### LINKED ISSUE #1423: Add table aws_cloudtrail_event_data_store
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.
[Go Docs](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudtrail#Client.GetEventDataStore)
[Terraform Docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudtrail_event_data_store)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_cloudtrail_event_data_store []

PRETEST: tests/aws_cloudtrail_event_data_store

TEST: tests/aws_cloudtrail_event_data_store
Running terraform
data.aws_region.alternate: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Reading...
data.aws_region.primary: Reading...
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_region.primary: Read complete after 0s [id=us-east-1]
data.aws_caller_identity.current: Read complete after 2s [id=632902152528]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_cloudtrail_event_data_store.named_test_resource will be created
  + resource "aws_cloudtrail_event_data_store" "named_test_resource" {
      + arn                            = (known after apply)
      + id                             = (known after apply)
      + multi_region_enabled           = true
      + name                           = "turbottest23807"
      + organization_enabled           = false
      + retention_period               = 2555
      + tags_all                       = (known after apply)
      + termination_protection_enabled = true

      + advanced_event_selector {
          + name = "Log all DynamoDB PutEvent actions for a specific DynamoDB table"

          + field_selector {
              + ends_with       = (known after apply)
              + equals          = [
                  + "AWS::DynamoDB::Table",
                ]
              + field           = "resources.type"
              + not_ends_with   = (known after apply)
              + not_equals      = (known after apply)
              + not_starts_with = (known after apply)
              + starts_with     = (known after apply)
            }
          + field_selector {
              + ends_with       = (known after apply)
              + equals          = [
                  + "Data",
                ]
              + field           = "eventCategory"
              + not_ends_with   = (known after apply)
              + not_equals      = (known after apply)
              + not_starts_with = (known after apply)
              + starts_with     = (known after apply)
            }
          + field_selector {
              + ends_with       = (known after apply)
              + equals          = [
                  + "PutItem",
                ]
              + field           = "eventName"
              + not_ends_with   = (known after apply)
              + not_equals      = (known after apply)
              + not_starts_with = (known after apply)
              + starts_with     = (known after apply)
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id    = "632902152528"
  + aws_partition = "aws"
  + region_name   = "us-east-1"
  + resource_aka  = (known after apply)
  + resource_name = "turbottest23807"
aws_cloudtrail_event_data_store.named_test_resource: Creating...
aws_cloudtrail_event_data_store.named_test_resource: Creation complete after 5s [id=arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/4b19aea8-fc7b-4f50-9863-bee5a24d0d88]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 44, in data "null_data_source" "resource":
  44: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "632902152528"
aws_partition = "aws"
region_name = "us-east-1"
resource_aka = "arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/4b19aea8-fc7b-4f50-9863-bee5a24d0d88"
resource_name = "turbottest23807"

Running SQL query: test-hydrate-query.sql
[
  {
    "advanced_event_selectors": [
      {
        "FieldSelectors": [
          {
            "EndsWith": null,
            "Equals": [
              "Data"
            ],
            "Field": "eventCategory",
            "NotEndsWith": null,
            "NotEquals": null,
            "NotStartsWith": null,
            "StartsWith": null
          },
          {
            "EndsWith": null,
            "Equals": [
              "PutItem"
            ],
            "Field": "eventName",
            "NotEndsWith": null,
            "NotEquals": null,
            "NotStartsWith": null,
            "StartsWith": null
          },
          {
            "EndsWith": null,
            "Equals": [
              "AWS::DynamoDB::Table"
            ],
            "Field": "resources.type",
            "NotEndsWith": null,
            "NotEquals": null,
            "NotStartsWith": null,
            "StartsWith": null
          }
        ],
        "Name": "Log all DynamoDB PutEvent actions for a specific DynamoDB table"
      }
    ],
    "name": "turbottest23807"
  }
]
✔ PASSED

Running SQL query: test-list-call-query.sql
[
  {
    "event_data_store_arn": "arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/4b19aea8-fc7b-4f50-9863-bee5a24d0d88",
    "name": "turbottest23807"
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/4b19aea8-fc7b-4f50-9863-bee5a24d0d88"
    ],
    "title": "turbottest23807"
  }
]
✔ PASSED

POSTTEST: tests/aws_cloudtrail_event_data_store

TEARDOWN: tests/aws_cloudtrail_event_data_store

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  name,
  event_data_store_arn,
  status,
  created_timestamp,
  multi_region_enabled,
  organization_enabled,
  termination_protection_enabled
from
  aws_cloudtrail_event_data_store;
+-----------------------+-----------------------------------------------------------------------------------------------+---------+---------------------------+----------------------+----------------------+--------------------------------+
| name                  | event_data_store_arn                                                                          | status  | created_timestamp         | multi_region_enabled | organization_enabled | termination_protection_enabled |
+-----------------------+-----------------------------------------------------------------------------------------------+---------+---------------------------+----------------------+----------------------+--------------------------------+
| turbottest23807       | arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/4b19aea8-fc7b-4f50-9863-bee5a24d0d88 | ENABLED | 2022-11-28T16:38:35+05:30 | true                 | false                | true                           |
| turbottest51242       | arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/03543d1c-054c-4016-95ae-1f8faef24485 | ENABLED | 2022-11-28T16:27:47+05:30 | true                 | false                | true                           |
| turbottest37839       | arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/3cde6043-35f1-4487-8f12-7dd097322bc6 | ENABLED | 2022-11-28T16:34:09+05:30 | true                 | false                | true                           |
| test-event-data-store | arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/5c5ae735-06f1-40b3-84d5-d9838162b1d1 | ENABLED | 2022-11-24T17:34:55+05:30 | true                 | false                | true                           |
| turbottest86634       | arn:aws:cloudtrail:us-east-1:632902152528:eventdatastore/8ed78734-cebd-472c-911a-37eb4c690d52 | ENABLED | 2022-11-28T16:22:43+05:30 | true                 | false                | true                           |
+-----------------------+-----------------------------------------------------------------------------------------------+---------+---------------------------+----------------------+----------------------+--------------------------------+

> select
  name,
  event_data_store_arn,
  status,
  created_timestamp,
  multi_region_enabled,
  organization_enabled,
  termination_protection_enabled
from
  aws_cloudtrail_event_data_store
where
  status <> 'ENABLED';
+------+----------------------+--------+-------------------+----------------------+----------------------+--------------------------------+
| name | event_data_store_arn | status | created_timestamp | multi_region_enabled | organization_enabled | termination_protection_enabled |
+------+----------------------+--------+-------------------+----------------------+----------------------+--------------------------------+
+------+----------------------+--------+-------------------+----------------------+----------------------+--------------------------------+

> select
  name,
  event_data_store_arn,
  status,
  created_timestamp,
  multi_region_enabled,
  organization_enabled,
  termination_protection_enabled
from
  aws_cloudtrail_event_data_store
where
  not termination_protection_enabled;
+------+----------------------+--------+-------------------+----------------------+----------------------+--------------------------------+
| name | event_data_store_arn | status | created_timestamp | multi_region_enabled | organization_enabled | termination_protection_enabled |
+------+----------------------+--------+-------------------+----------------------+----------------------+--------------------------------+
+------+----------------------+--------+-------------------+----------------------+----------------------+--------------------------------+

```
</details>

### CODE REVIEW

File: aws/plugin.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_cloudsearch_domain":                                       tableAwsCloudSearchDomain(ctx),
 			"aws_cloudtrail_trail":                                         tableAwsCloudtrailTrail(ctx),
 			"aws_cloudtrail_trail_event":                                   tableAwsCloudtrailTrailEvent(ctx),
+			"aws_cloudtrail_event_data_store":                              tableAwsCloudtrailEventDataStore(ctx),
```
Comment (misraved) [inline aws/plugin.go:87]: Could you please sort this list of tables?

File: aws/table_aws_cloudtrail_trail_event_data_store.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+//// TABLE DEFINITION
+
+func tableAwsCloudtrailEventDataStore(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_cloudtrail_event_data_store",
+		Description: "AWS CloudTrail Event Data Store",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AnyColumn([]string{"event_data_store_arn"}),
+			Hydrate: getCloudTrailEventDataStore,
```
Comment (misraved) [inline aws/table_aws_cloudtrail_trail_event_data_store.go:23]:
```suggestion
			KeyColumns: plugin.AnyColumn([]string{"event_data_store_arn"}),
			Hydrate:        getCloudTrailEventDataStore,
```

File: aws/table_aws_cloudtrail_trail_event_data_store.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+			{
+				Name:        "organization_enabled",
+				Description: "Indicates that an event data store is collecting logged events for an organization.",
+				Type:        proto.ColumnType_BOOL,
+				Hydrate:     getCloudTrailEventDataStore,
+				Transform:   transform.FromField("OrganizationEnabled"),
+			},
+			{
+				Name:        "retention_period",
+				Description: "This field is being deprecated. The retention period, in days.",
```
Comment (misraved) [inline aws/table_aws_cloudtrail_trail_event_data_store.go:70]: If this field is being deprecated, does it make sense to keep it?

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!!

File: aws/table_aws_cloudtrail_trail_event_data_store.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
+package aws
```
Comment (cbruno10) [inline aws/table_aws_cloudtrail_trail_event_data_store.go:1]: The filename looks incorrect, I think it should be `aws/table_aws_cloudtrail_event_data_store.go`

File: aws/table_aws_cloudtrail_trail_event_data_store.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCloudtrailEventDataStore(_ context.Context) *plugin.Table {
```
Comment (cbruno10) [inline aws/table_aws_cloudtrail_trail_event_data_store.go:17]:
```suggestion
func tableAwsCloudTrailEventDataStore(_ context.Context) *plugin.Table {
```

The service name is `CloudTrail`, so `Trail` should be capitalized in the function name.

File: docs/tables/aws_cloudtrail_event_data_store.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# Table: aws_cloudtrail_event_data_store
+
+AWS CloudTrail Event data store stores data for up to seven years, or 2557 days. By default, event data is retained for 2557 days, and termination protection is enabled for an event data store.
```
Comment (cbruno10) [inline docs/tables/aws_cloudtrail_event_data_store.md:3]:
```suggestion
AWS CloudTrail event data stores retain data for up to seven years, or 2557 days. By default, event data is retained for 2557 days, and termination protection is enabled for an event data store.
```

File: docs/tables/aws_cloudtrail_event_data_store.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  status,
+  created_timestamp,
+  multi_region_enabled,
+  organization_enabled,
+  termination_protection_enabled
+from
+  aws_cloudtrail_event_data_store;
+```
+
+### List cloudtrail event stores which are not enabled
```
Comment (cbruno10) [inline docs/tables/aws_cloudtrail_event_data_store.md:22]:
```suggestion
### List event data stores which are not enabled
```

We don't need the service name (unless it's to disambiguate between resource names together), and we should include the full resource name

File: docs/tables/aws_cloudtrail_event_data_store.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  multi_region_enabled,
+  organization_enabled,
+  termination_protection_enabled
+from
+  aws_cloudtrail_event_data_store
+where
+  status <> 'ENABLED';
+```
+
+### List cloudtrail event stores with termination protection disabled
```
Comment (cbruno10) [inline docs/tables/aws_cloudtrail_event_data_store.md:39]:
```suggestion
### List event data stores with termination protection disabled
```

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see review comments, thanks!

File: aws/table_aws_cloudtrail_event_data_store.go [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the event data store.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "event_data_store_arn",
```
Comment (cbruno10) [inline aws/table_aws_cloudtrail_event_data_store.go:36]:
```suggestion
				Name:        "arn",
```

I think we usually normalize the `arn` columns to just be `arn` (so it's easy to find and so our mod controls work well with them).

File: aws/table_aws_cloudtrail_event_data_store.go [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+		plugin.Logger(ctx).Info("aws_cloudtrail_event_data_store.listCloudTrailEventDataStores", "client_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(1000)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
```
Comment (cbruno10) [inline aws/table_aws_cloudtrail_event_data_store.go:121]: I think in the later Steampipe CLI versions if someone runs a query with `limit 0`, the hydrate functions aren't entered, so we may not need this check, so we could simplify to:
```
		if limit < maxLimit {
			maxLimit = limit
		}
```

Can you please test and confirm this?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see review comments, thanks!

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**